applicationinsights-android/src/main/java/com/microsoft/applicationinsights/internal/TelemetryContext.java

```diff
@@ -274,24 +284,35 @@ protected static void setAppContext(Context appContext) {
     /**
      * Sets the user context
      */
-    protected static void setUserContext() {
-        String userId = TelemetryContext.settings.getString(TelemetryContext.USER_ID_KEY, null);
-        String userAcq = TelemetryContext.settings.getString(TelemetryContext.USER_ACQ_KEY, null);
+    public static void setUserContext(String userId) {
+        String userAcq = null;
+
+        // No custom user Id is given, so get this info from settings
+        if(userId == null){
+            userId = TelemetryContext.settings.getString(TelemetryContext.USER_ID_KEY, null);
+            userAcq = TelemetryContext.settings.getString(TelemetryContext.USER_ACQ_KEY, null);
+        }
 
-        if (userId == null || userAcq == null) {
+        // Generate new user Id if needed
+        if (userId == null) {
             userId = UUID.randomUUID().toString();
-            userAcq = Util.dateToISO8601(new Date());
+        }
 
-            SharedPreferences.Editor editor = TelemetryContext.settings.edit();
-            editor.putString(TelemetryContext.USER_ID_KEY, userId);
-            editor.putString(TelemetryContext.USER_ACQ_KEY, userAcq);
-            editor.apply();
+        // Generate new acquisition date if needed
+        if(userAcq == null) {
```

Comment:
let's set this if/when we generate the user ID. If user Id is set otherwise we can't accurately report the acquisition date.